app/src/main/java/ch/admin/bag/dp3t/checkin/checkinflow/CheckInFragment.kt

```diff
@@ -0,0 +1,95 @@
+package ch.admin.bag.dp3t.checkin.checkinflow
+
+import android.os.Bundle
+import android.view.ContextThemeWrapper
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.view.ViewGroup.LayoutParams.WRAP_CONTENT
+import android.widget.LinearLayout
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.activityViewModels
+import ch.admin.bag.dp3t.R
+import ch.admin.bag.dp3t.checkin.CrowdNotifierViewModel
+import ch.admin.bag.dp3t.checkin.models.ReminderOption
+import ch.admin.bag.dp3t.checkin.utils.*
+import ch.admin.bag.dp3t.databinding.FragmentCheckInBinding
+import com.google.android.material.button.MaterialButton
+
+class CheckInFragment : Fragment() {
+
+	companion object {
+		val TAG = CheckInFragment::class.java.canonicalName
+
+		@JvmStatic
+		fun newInstance(): CheckInFragment {
+			return CheckInFragment()
+		}
+	}
+
+	private val viewModel: CrowdNotifierViewModel by activityViewModels()
+
+	override fun onCreate(savedInstanceState: Bundle?) {
+		super.onCreate(savedInstanceState)
+		checkIfAutoCheckoutHappened()
+	}
+
+	override fun onStart() {
+		super.onStart()
+		checkIfAutoCheckoutHappened()
+	}
+
+	private fun checkIfAutoCheckoutHappened() {
+		if (viewModel.checkInState == null) {
+			popBackToHomeFragment()
+		}
+	}
+
+	override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+		return FragmentCheckInBinding.inflate(inflater).apply {
+			val venueInfo = viewModel.checkInState?.venueInfo ?: return root
+
+			titleTextview.text = venueInfo.title
+			subtitleTextview.text = venueInfo.getSubtitle()
+			checkInButton.setOnClickListener {
+				val checkInTime = System.currentTimeMillis()
```

Comment:
I'd move the checkin logic (everything except the `popBackToHomeFragment()`) into a separate method.